### DISCUSSION TITLE
Ability to test maps in-Editor / a huge overlook with editor mode as a long time worldedit user.

### DISCUSSION THREAD

**Discussion by AliveJy** [Mojang/minecraft-editor#64] [Feedback]
So currently, editor mode is a great start for what is to come with maps in bedrock. as for a long time, people have been using minecraft java's worldedit to make amazing builds with all of the world edit commands and features (seriously, world edit is a life saver, you might not notice it until you actually start using world edit for creating large-idea maps), then after making the world in java, they would convert them to bedrock using bedrock map conversion tools like chunker.app, which works just fine.

But that one thing... that one major thing that's missing is... being to actually be able to test the map, and yes. sure you could just make a map without testing **ANY** of the features in it, then when you actually test the map after converting it to a normal world, you'll realize you made a mess, because you couldn't actually test it because all entities don't move, some commands are not avaible, even survival mode is not available and most normal minecraft mechanics are not present.

Most of the time when I work on my maps, I build them and do the commands at the same time when building, like when I feel like it, I use world edit to make cool mountains and texturing, but then a second later I would be testing the mechanics and working using the command blocks, then return back to building... which most people do... so when you can't actually test the maps, the maps would flop.

Consider looking into this maybe?

**Comment by SmokeyStack** [comment]
Entities not moving is the default, you can unpause it in the menu bar. Switching to cross hair mode is how you test things. Use a command block as workaround to switch gamemode

**Comment by hmm-monke** [comment]
can you switch gamemodes with the chat?

**Reply by SmokeyStack** [reply]
https://github.com/Mojang/minecraft-editor/discussions/40

**Comment by ihategravel2** [comment]
> Entities not moving is the default, you can unpause it in the menu bar. Switching to cross hair mode is how you test things. Use a command block as workaround to switch gamemode


You can also change your gamemode by just going to the Pause Menu.

**Comment by G01dY05h1** [comment]
To add on to this, you should also be able to switch your permission level on the fly with a key combo.

Since you now need to be in editor mode and use a key combo to pause the game, it takes so long to, say, switch from op, to visitor, and then back again. With a key combo you could easily just switch permissions with the press of a button.

This is great for testing border blocks for example without having to assume they are in the right spot before switching modes and going to fix it and switching back.
